### PR TITLE
HPCC-23313 Add shebang to the shell scripts which can be run individually

### PR DESCRIPTION
**Pull request by xwang2713**
Change default python to python3 in shebang section
Add "#!/bin/sh" for executable shell scripts

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

@Michael-Gardner please review

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

CE Platform: http://10.240.32.243/view/CentOS8/job/CE-CentOS8/CE=centos-8.0-x86_64/
CE Plugins: http://10.240.32.243/view/CentOS8/job/CE-Plugins-CentOS8/CE=centos-8.0-x86_64/1/console
  http://10.240.32.243/view/CentOS8/job/CE-Plugins-CentOS8/CE=centos-8.0-x86_64/3/console
  V8 failed:https://track.hpccsystems.com/browse/HPCC-23447
  MySQL failed: https://track.hpccsystems.com/browse/HPCC-23431

LN Platform: http://10.240.32.243/view/CentOS8/job/LN-CentOS8/

### CODE REVIEW

**Comment by xwang2713** [comment]
CentOS 8 development setup: http://10.240.34.10/wiki/index.php/New_OS_Tests#Build_Server_Pre.C3.A5requisites

File: initfiles/examples/embed/python_cat.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 from __future__ import print_function
```
Comment (richardkchapman) [inline initfiles/examples/embed/python_cat.py]: You don't need this any more if you are using python3

File: testing/regress/ecl-test [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (richardkchapman) [inline testing/regress/ecl-test]: Have you tested that this works with python3?

**Comment by richardkchapman** [comment]
This should target master for 7.8

Also please correct the commit title so it links to a Jira issue

File: initfiles/examples/embed/python_cat.py (hunk 1 above)
Comment (xwang2713) [reply initfiles/examples/embed/python_cat.py]: But on my CentOS 8 get error "ambiguous python shebang"

**Comment by xwang2713** [comment]
I change back from "#! /usr/bin/python3" to "#!/usr/bin/python".  
On build server we set python3 as default python: sudo alternatives --config python

Also should we only support python3 on CentOS 8 since python2 retired on January 1, 2020?

**Comment by richardkchapman** [comment]
@Michael-Gardner Please review. I assume that sh (as opposed to bash) is enough for all these?

**Comment by xwang2713** [comment]
@Michael-Gardner, set default python to python3 seems not enough. CentOS 8  follows "Packages" changed since Fedora 31: i.e. "Packages in Fedora MUST NOT use /usr/bin/python. Instead packages for Python 3 MUST use /usr/bin/python3".  For compatibility for other distros I add %undefine __brp_mangle_shebangs" to skip shebang process in rpm generation. Here is the reference docs:  https://docs.fedoraproject.org/en-US/packaging-guidelines/#_shebang_lines

File: CMakeLists.txt [hunk 3]
```diff
@@ -392,7 +392,8 @@ if(TOP_LEVEL_PROJECT)
                 "%define _use_internal_dependency_generator 0
                 %define __getdeps()while read file; do /usr/lib/rpm/rpmdeps -%{1} ${file} | %{__grep} -v libantlr3c.so | %{__grep} -v libtbb.so ; done | /bin/sort -u
                 %define __find_provides /bin/sh -c '%{__getdeps P}'
-                %define __find_requires /bin/sh -c '%{__getdeps R}'")
+                %define __find_requires /bin/sh -c '%{__getdeps R}
```
Comment (richardkchapman) [inline CMakeLists.txt]: Should there be ' on the end of this line?

**Comment by xwang2713** [comment]
@Michael-Gardner I opened JIRA to drop Python2 :  https://track.hpccsystems.com/browse/HPCC-23740

**Comment by Michael-Gardner** [comment]
@xwang2713 for the most part it looks ok.  line 395 where the %define code is does look like it needs to be terminated by a quotation?  I'll build this branch and test it out.

**Comment by xwang2713** [comment]
Let me double check

**Comment by xwang2713** [comment]
@Michael-Gardner you are right. I will fix and upload again

**Comment by Michael-Gardner** [comment]
Fetching and testing

**Comment by xwang2713** [comment]
Also rebase with master to get the fix for mysql embed plugin

File: CMakeLists.txt (hunk 3 above)
Comment (xwang2713) [reply CMakeLists.txt]: Fixed

**Comment by richardkchapman** [comment]
@xwang2713 Seems to have stalled - what is this waiting for?

**Comment by Michael-Gardner** [comment]
It was looking good to me after that one change.  Do you need me to test this again Ming?

**Comment by xwang2713** [comment]
I can't build this anymore. Will rework for 7.8.10 branch

**Pull request closed without merging**